pkg/controller/signup.go

```diff
@@ -90,11 +90,11 @@ func (s *Signup) UpdateVerificationHandler(ctx *gin.Context) {
 		"-", "")
 	countryCode := r.Replace(phone.CountryCode)
 	phoneNumber := r.Replace(phone.PhoneNumber)
-	matchedCode, err := regexp.MatchString("[^0-9]+", countryCode)
-	matchedPhone, err := regexp.MatchString("[^0-9]+", phoneNumber)
+	countryValid, _ := regexp.MatchString(`^\+?[0-9]+$`, countryCode)
+	phoneValid, _ := regexp.MatchString(`^[0-9]+$`, phoneNumber)
```

Comment:
Could you please add unit tests verifying the regex?
Ideally, extract the verification logic to functions and run a set of test cases against it. Thanks to the tests it will be easier to understand which numbers/country codes are valid and  which not.

Reply:
There's already a test for this - https://github.com/codeready-toolchain/registration-service/blob/master/pkg/controller/signup_test.go#L460

I'm not going to touch the structure of the code in this PR, that is planned for a separate PR.  This PR is intended as a targeted set of changes to enable the e2e tests to pass, and due to the restrictions of the e2e test setup not being capable of executing test across 3 branches this PR will need to be merged in order for the corresponding PRs in toolchain-e2e and host-operator to pass.

Reply:
Ok, my point was that you changed the regex, then I would expect that we have a test case that covers this change, otherwise, we don't know if the regex is finally OK-ish or requires some additional changes. Apart from that, existing unit test would ensure that any future refactoring changes won't break anything.
In addition, such tests would show what the regex expects, what is valid and what not - it would be really helpful for dev who are not strong at regexes ;-)

Reply:
I changed the regex because an existing test (from master) was failing, however when running the tests from the master branch the tests pass.... so really weird behaviour.  In any case, the regex was wrong, it was searching for any match even if the string as a whole contained invalid characters.  If I had more time I'd dig deeper into why the tests seem to be passing in master however I've not had a spare moment to look at this as of yet.

Reply:
@sbryzak this regex should only match strings that are anything other than the numbers 0-9. in the regex you've provided you're doing the reverse. whichever way is fine, but i don't understand how/why the regex was broken. you can test it out here: https://regex101.com/

Reply:
To be honest, I would rather prefer having a large set of strings that are executed against the regex instead of using third-party service for verifying that the regex works. We could either store them as global variable and use them is special test cases or move them into separated function doing the only function that will verify the validity and test the string against the function. Something similar as we have for username transformation: https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/usersignup/usersignup_controller_test.go#L2010-L2035

Reply:
Fully agree with @MatousJobanek. We have to have a list of explicit assertions testing different variants of phone numbers.
We can't say that this code works until we have those tests in place. We wouldn't have this argument regarding whether the new/old regex is correct or not if we had proper tests in place.

Reply:
@tinakurian btw, Shane is correct with his fix - the original regex was wrong: `"[^0-9]+"`
* for specifying regex, we don't use the normal "double apostrophe" but the backquote (but it's just a detail)
* AFAIK the char `^` in the square brackets means negation so your regex accepted all strings that didn't contain a number
* if you want to accept only the numbers you need to use: `[0-9]+`

Reply:
@MatousJobanek i fully intended the regex to match on everything other than numbers. so it was doing what i expected and wanted. as for the third party tool, i had only suggested it for testing out what exactly the regex works against in terms of understanding it since @sbryzak had mentioned that it was broken.. again, i had intended it to do what it does - match on characters that aren't numbers..